pkg/controllers/gateway/params.go

```diff
@@ -0,0 +1,89 @@
+package gateway
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+type Params struct {
+}
+
+var defaultParams Params = Params{}
+
+type InvalidParamsError struct {
+	message string
+}
+
+var _ error = &InvalidParamsError{}
+
+// Error implements error
+func (e *InvalidParamsError) Error() string {
+	return e.message
+}
+
+func IsInvalidParamsError(err error) (is bool) {
```

Comment:
I always wonder if the default for these should be

```
if err == nil{
return false
}
```

Reply:
That's an interesting question, because it would be possible to define
```go
var err1 *InvalidParamsError = nil
var err2 error = nil
```

Both err1 and err2 would be `nil` but passing err1 to `IsInvalidParamsError` would return true, and passing err2 would return false.
I think it makes sense to leave it like this and leave the nil check responsibility to the caller of `IsInvalidParamsError`